src/libertem/io/dataset/frms6.py

```diff
@@ -153,6 +170,28 @@ def _read_file_header(path):
     return header
 
 
+def _get_sig_shape(path, bin_factor):
+    path, ext = os.path.splitext(path)
+    if ext == ".hdr":
+        pattern = "%s_*.frms6" % path
+    elif ext == ".frms6":
+        pattern = "%s*.frms6" % (
+            re.sub(r'[0-9]+$', '', path)
+        )
+    else:
+        raise DataSetException("unknown extension: %s" % ext)
+    filenames = list(sorted(glob(pattern)))
+    headers = [
+        _read_file_header(path)
+        for path in filenames
+    ]
+    header = headers[0]
```

Comment:
No need to read all headers if only the first one is used - how about `header = _read_file_header(filenames[0])` or something like that?